src/engine/SpatialJoinAlgorithms.cpp

```diff
@@ -89,6 +89,15 @@ std::pair<util::geo::I32Box, size_t> SpatialJoinAlgorithms::libspatialjoinParse(
         "prefilter-disabled-by-bounding-box-area", true);
   }
 
+  // If the input is smaller than one batch for every thread, reduce the number
+  // of threads accordingly to avoid spawning threads that will never be used.
+  static constexpr auto batchSize =
+      ad_utility::detail::parallel_wkt_parser::WKT_PARSER_BATCH_SIZE;
```

Comment:
Can you add a `static_assert` that the batch size is non-zero to ensure we don't have underflows/division by zero?